File: Web service/ADO-NET/chilaadblib/championsleague.cs

#region includes
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Collections;
using System.Data;
using System.Data.SqlClient;
#endregion

namespace chilaaadodblib
{
    public class championsleague
    {
         #region Data

        public SqlConnection cnn;

        #endregion

        #region Constructor-Destructor
        public championsleague(String dbname, String uid, String pass)
        {
    
            cnn = new SqlConnection();
            cnn.ConnectionString = String.Format("Data Source=(local);Initial Catalog=ChampionsLeague;Integrated Security=True;User ID={1};Password={2};", dbname, uid, pass);
            cnn.Open();

        }

     ~championsleague()
        {
            try
            {
                if (cnn.State == ConnectionState.Open)
                    cnn.Close();

            }
            catch (Exception e) { Console.WriteLine(e.Message); }
        }
 #endregion

        #region GeneralPurposeDBMethods


        public DataSet RunSelect(SqlConnection cnn, String query)

        {
            DataSet ds = new DataSet();
            try
            {

                SqlDataAdapter da = new SqlDataAdapter();
                SqlCommand cmd = new SqlCommand(query, cnn);

                // Set the command to run
                da.SelectCommand = cmd;
                //fill the data set
                da.Fill(ds);
              
                //return the dataset
            }
            catch (Exception e) { Console.WriteLine(e.Message); }
            return ds;
        }

           public void TransactSQL(String[] sql, SqlConnection cnn)


        {

            SqlCommand cmd = new SqlCommand();
            SqlTransaction trans = cnn.BeginTransaction();

            cmd.Connection = cnn;
            try
            {
                // Assign transaction object for a pending local transaction
                cmd.Transaction = trans;

                for (int i = 0; i < sql.Length; i++)
                {
                    cmd.CommandText = sql[i];
                    cmd.ExecuteNonQuery();
                }
                //Commit the transaction
                trans.Commit();
            }
            catch (InvalidOperationException ex)
            {
                //MessageBox.Show(ex.Message); 
                trans.Rollback();
                throw ex;
            }
            catch (Exception ex)
            {
                //MessageBox.Show(ex.Message);
                trans.Rollback();
                throw ex;
            }

        }
        #endregion
        

        #region DBSpecificMethods

         public String AddPlayerStat(String mid, String pid, int goals){
                String pname = null;
             try{
                 String[] sql;
                 String s = String.Format("Select Team1, Team2 as TeamID from MatchSchedule WHERE MID = '{0}'",mid);
                 DataSet ds = RunSelect(cnn, s);
                 DataTable dt = ds.Tables[0];

                 String tid1 = dt.Rows[0][0].ToString();
                 String tid2 = dt.Rows[0][1].ToString();

                String q = String.Format("SELECT CASE WHEN EXISTS(SELECT * FROM TeamRoster where (TID = '{0}' OR TID = '{1}') AND PID = '{2}') THEN 'true' ELSE 'false' END AS Verify FROM TeamRoster", tid1,tid2,pid);
                ds = RunSelect(cnn, q);

                dt = ds.Tables[0];
                Boolean verify = Convert.ToBoolean(dt.Rows[0][0].ToString());
 
                String r = String.Format("Select TID from TeamRoster where PID = '{0}'",pid);
                ds = RunSelect(cnn, r);

                dt = ds.Tables[0];

                String tid = dt.Rows[0][0].ToString();
                
                 if(verify == true){
                Console.WriteLine("Verified"+verify);
                 sql = new String[2];
                
                     
                         sql[0]=String.Format("Insert Into MatchRoster(MID,PID,Goals) Values ('{0}','{1}',{2})", mid, pid, goals);
                         sql[1]=String.Format("Update TeamRoster Set Goals = (Select sum(Goals) from MatchRoster where PID = '{0}') WHERE PID = '{1}' AND TID = '{2}'", pid, pid, tid);
                      //   sql[2]=String.format("Select Pname from Player where PID = '{0}'",pid);

                     try{
                         TransactSQL(sql,cnn);
                         String p = String.Format("Select Pname from Player where PID = '{0}'",pid);
                         ds = RunSelect(cnn, p);

                         dt = ds.Tables[0];

                         pname = "The stats for" + dt.Rows[0][0].ToString() + "is updated";
                         }
                     catch (Exception ex) { Console.WriteLine(ex.Message); }
                  }
                else{
                pname = "Player did not play the match";

                }
                  
             }
              catch (SqlException ex) {
                    Console.WriteLine(ex.Message);
                } catch (Exception e) {
                    Console.WriteLine(e.Message); 
        
               }
              return pname;
                       
             }
 
       public List<String> ListPlayerStats(String pid) {
              List<String> rv = new List<String>();
             try{
              String a = String.Format("SELECT Pname, Goals FROM Player,TeamRoster WHERE Player.PID = '{0}' AND TeamRoster.PID = '{1}'",pid,pid);
                 
              DataSet ds = RunSelect(cnn, a);
              DataTable dt = ds.Tables[0];

              String line;
              for (int i = 0; i < dt.Rows.Count; i++)
              {

                  String player = dt.Rows[0][0].ToString();
                  int goals = Convert.ToInt32(dt.Rows[0][1].ToString());
                  
                  line = String.Format("{0,-15}\t{1,-1}\t{2,-1}\t{3,-10}", player, "-", goals, "goals");
                  rv.Add(line);
              }
                line = String.Format("{0,-15}\t{1,-20}\t{2,-5}", "Opponent","Date", "Goals");
                rv.Add(line);
                line = String.Format("{0,-15}\t{1,-20}\t{2,-5}", "---------------", "--------------------", "----");
                rv.Add(line);

                String c = String.Format("Select TID from TeamRoster where PID = '{0}'",pid);
                 ds = RunSelect(cnn,c);
                 dt = ds.Tables[0];

                String tid = dt.Rows[0][0].ToString();

               String d = String.Format("Select MID, MDate from MatchSchedule where Team1= '{0}' OR Team2= '{1}'",tid,tid);
                 ds = RunSelect(cnn,d);
                 dt = ds.Tables[0];


                 for (int i = 0; i < dt.Rows.Count; i++)
                 {
                     String mid = dt.Rows[i][0].ToString();
                     String mdate = dt.Rows[i][1].ToString();

                     String e = String.Format("Select Goals from MatchRoster where MID = '{0}' AND PID = '{1}'", mid, pid);
                     ds = RunSelect(cnn, e);
                     DataTable dt1 = ds.Tables[0];

                     int goal = 0;
                     for (int j = 0; j < dt1.Rows.Count; j++)
                     {
                         goal = Convert.ToInt32(dt1.Rows[j][0].ToString());
                     }
                     
                     String f = String.Format("(Select Team1 as TeamID from MatchSchedule where Team2='{0}' AND MID = '{1}')Union All (Select Team2 as TeamID from MatchSchedule where Team1='{2}' AND MID = '{3}')", tid, mid, tid, mid);
                     ds = RunSelect(cnn, f);
                     DataTable dt2 = ds.Tables[0];
                     String teamid = dt2.Rows[0][0].ToString();

                     String g = String.Format("Select TName from Team where TID = '{0}'", teamid);
                     ds = RunSelect(cnn, g);
                     DataTable dt3 = ds.Tables[0];
                     String tname = dt3.Rows[0][0].ToString();

                     line = String.Format("{0,-15}\t{1,-20}\t{2,-5}", tname, mdate, goal);
                     rv.Add(line);

                 }
               } catch (SqlException ex) {
                    Console.WriteLine(ex.Message);
                } catch (Exception e) {
                    Console.WriteLine(e.Message); 
                }
               
               
               return rv;
            }

         public List<String> ListTeamPoints(String tid) {
                List<String> retval = new List<String>();
                try{
                String l = String.Format("Select Tname from Team where TID = '{0}'",tid);
                 
                DataSet ds = RunSelect(cnn, l);
                DataTable dt = ds.Tables[0];
   
                String line = String.Format("{0,-12}\t{1,-3}\t{2,-3}\t{3,-3}\t{4,-3}\t{5,-3}\t{6,-3}\t","Team","W","T","L","G","A","P");
                retval.Add(line);
                line = String.Format("{0,-12}\t{1,-3}\t{2,-3}\t{3,-3}\t{4,-3}\t{5,-3}\t{6,-3}\t", "------------", "---", "---", "---", "---", "---", "---");
                retval.Add(line);

                for (int i = 0; i < dt.Rows.Count; i++)
                {
                    String name = dt.Rows[0][0].ToString();

                    String win = String.Format("Select Count(*) as Won from Match M1,Match M2 where M1.TID = '{0}' AND M1.MID = M2.MID AND M1.Goals>M2.Goals ", tid);

                    ds = RunSelect(cnn, win);
                    dt = ds.Tables[0];
                    String won = dt.Rows[0][0].ToString();

                    String tie = String.Format("Select Count(*) as Ties from Match M1,Match M2 where M1.TID = '{0}' AND M1.MID = M2.MID AND M1.Goals = M2.Goals AND M1.TID<>M2.TID ", tid);

                    ds = RunSelect(cnn, tie);
                    dt = ds.Tables[0];
                    String ties = dt.Rows[0][0].ToString();

                    String loss = String.Format("Select Count(*) as Lost from Match M1,Match M2 where M1.TID = '{0}' AND M1.MID = M2.MID AND M1.Goals<M2.Goals", tid);

                    ds = RunSelect(cnn, loss);
                    dt = ds.Tables[0];
                    String lost = dt.Rows[0][0].ToString();


                    String go = String.Format("Select Sum(Goals) as Goals from TeamRoster where TID = '{0}' ", tid);

                    ds = RunSelect(cnn, go);
                    dt = ds.Tables[0];
                    String goals = dt.Rows[0][0].ToString();

                    String all = String.Format("Select sum(Goals) from MatchRoster where MID IN (Select MID from MatchSchedule where Team1 = '{0}' OR Team2 = '{1}') AND\n" +
   "PID IN (Select PID from TeamRoster where TID IN ((Select Team1 from MatchSchedule WHERE Team2 = '{2}')Union (Select Team2 from MatchSchedule where Team1 = '{3}')))", tid, tid, tid, tid);

                    ds = RunSelect(cnn, all);
                    dt = ds.Tables[0];
                    String goalsall = dt.Rows[0][0].ToString();

                    int points = (Convert.ToInt32(won) * 3) + (Convert.ToInt32(ties) * 1);

                    line = String.Format("{0,-12}\t{1,-3}\t{2,-3}\t{3,-3}\t{4,-3}\t{5,-3}\t{6,-3}\t", name, won, ties, lost, goals, goalsall, points);
                    retval.Add(line);
                }
             } catch (SqlException ex) {
                    Console.WriteLine(ex.Message);
                } catch (Exception e) {
                    Console.WriteLine(e.Message); 
                }
             
             return retval;

         }
        #endregion

    }


        
    }

File: chilaadblib/src/chilaadblib/championsleague.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package chilaadblib;

import java.sql.*; //DB
import java.util.*; //for lists
import java.time.LocalDate;
/**
 *
 * @author achila
 */
public class championsleague {
    
    Connection mcn;
    
    public championsleague(String dbname, String uid, String pass) {

        try {

            setConnection(dbname, uid, pass);

        } catch (Exception e) {
            e.printStackTrace();
            return;
        }
    }
    


public void setConnection(String db, String uid, String pass) {
        try {
            //Per microsoft documentation no need to load the driver explicitly. Get connection does that on our behalf.
            // See http://msdn.microsoft.com/en-us/library/ms378526.aspx

            Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
           String connectionUrl = "jdbc:sqlserver://localhost\\MSSQLSERVER;"
           + "databaseName=" + db + ";user=" + uid + ";password="+ pass + ";";

            mcn = DriverManager.getConnection(connectionUrl);
        } catch (SQLException ex) {
            System.out.println(ex.getMessage());
            System.out.println("Error code:" + ex.getErrorCode());//ex.printStackTrace();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

public boolean IsConnected() {
        return (mcn != null ? true : false);
    }

    private String fixStringFromJDBC(String s) {
        if (s == null) {
            s = "";
        }
        if (s.equals(" ")) {
            s = "";
        }
        s = s.replace('`', '\'');
        return s;
    }

   /* private void TransactSQL(String[] sql) {
        
        int n = 0;
        try {
            
            //System.out.println(mcn.getAutoCommit());
            mcn.setAutoCommit(false);
            PreparedStatement pst = null;

            for (int i = 0; i < sql.length; i++) {
                 pst = mcn.prepareStatement(sql[i]);
                 pst.executeQuery();
            }
            //mcn.rollback();
            mcn.commit();
        } catch (SQLException ex) {
            try {
                //rollback if there is an error
                mcn.rollback();
                ex.printStackTrace();
            } catch (Exception e) {
                e.printStackTrace();
            }
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            try {
                if (pst != null) {
                    pst.close();
                }
            } catch (SQLException ex) {
                ex.printStackTrace();
            }
        }
        
    }*/
    
    public String AddPlayerStat(String mid, String pid, int goals){
         String pname = null;
        try{
          String sql[] = null; 
          
        PreparedStatement pst = mcn.prepareStatement("Select Team1, Team2 as TeamID from MatchSchedule WHERE MID = ?");
            pst.setString(1, mid);
            
            ResultSet rs = pst.executeQuery();
            rs.next();
            String tid1 = rs.getString(1);
            String tid2 = rs.getString(2);
          
        PreparedStatement pst1 = mcn.prepareStatement(
                "SELECT CASE WHEN EXISTS(SELECT * FROM TeamRoster where (TID = ? OR TID = ?) AND PID = ?) THEN 'true' ELSE 'false' END AS Verify FROM TeamRoster ");
                    
 
        
        pst1.setString(1,tid1);
        pst1.setString(2,tid2);
        pst1.setString(3,pid);
        ResultSet r = pst1.executeQuery();
        r.next();
        Boolean verify = r.getBoolean(1);
        //System.out.println(verify);
        PreparedStatement ps = mcn.prepareStatement("Select TID from TeamRoster where PID = ?");
        ps.setString(1, pid);
        ResultSet r1 = ps.executeQuery();
        r1.next();
        String tid = r1.getString(1);
        if(verify = true){
              System.out.println("Verified"+verify);
        //  sql = new String[3];
              try{
              mcn.setAutoCommit(false);
            PreparedStatement s1 = mcn.prepareStatement("Insert Into MatchRoster(MID,PID,Goals) Values (?,?,?)");
            s1.setString(1, mid);
            s1.setString(2, pid);
            s1.setInt(3, goals);
            PreparedStatement s2 = mcn.prepareStatement("Update TeamRoster Set Goals = (Select sum(Goals) from MatchRoster where PID = ?) WHERE PID = ? AND TID = ?");
            s2.setString(1, pid);
          //  s2.setInt(1, goals);
            s2.setString(2, pid);
            s2.setString(3, tid);
            PreparedStatement s3 = mcn.prepareStatement("Select Pname from Player where PID = ?");
            s3.setString(1, pid);
            s1.execute();
            s2.executeUpdate();
           ResultSet r3 = s3.executeQuery();
           r3.next();
          pname = "The stats for "+r3.getString(1)+" is updated";
          
               mcn.commit();
        } catch (SQLException ex) {
            try {
                //rollback if there is an error
                mcn.rollback();
                ex.printStackTrace();
            } catch (Exception e) {
                e.printStackTrace();
            }
        } catch (Exception e) {
            e.printStackTrace();
        } 
           
         //   sql[0]=String.format("Insert Into MatchRoster(MID,PID,Goals) Values ('%s','%s',%d);", mid, pid, goals);
          //  sql[1]=String.format("Update TeamRoster Set Goals = %d WHERE PID = '%s' AND TID = '%s';", goals, pid, tid);
           // sql[2]=String.format("Select Pname from Player where PID = '%s';",pid);
         //  TransactSQL(sql);
        }
        else{
            pname = "Player did not play the match";
        }
        
            
        } catch (SQLException ex) {
            ex.printStackTrace();
        
        } catch (Exception e) {
            e.printStackTrace();
            
       
        }
       return pname;
    }
    
    public List<String> ListPlayerStats(String pid) {
        List<String> rv = new ArrayList<String>();
        try {

            //CallableStatement st = mcn.prepareCall("{call GetPurchaseRequests(?) }");
/*            
             Statement s = mcn.createStatement();
             String sql = 
             String.format("SELECT Amount, Appcode, CodeType FROM PurchaseRequests WHERE AccountNo='%s'", accno)
             */
            PreparedStatement st = mcn.prepareStatement(
                    "SELECT Pname, Goals FROM Player,TeamRoster WHERE Player.PID = ? AND TeamRoster.PID = ?;");
            st.setString(1, pid);
            st.setString(2, pid);

            ResultSet rs = st.executeQuery();
            String line;
            while (rs.next()) {

                String player = rs.getString(1);
                int goals = rs.getInt(2);
          
                
                line = String.format("%-15s\t%-1s\t%-1d\t%-10s", player,"-", goals,"goals");
                rv.add(line);
            }
                line = String.format("%-15s\t%-20s\t%-5s", "Opponent","Date", "Goals");
                rv.add(line);
                line = String.format("%-15s\t%-20s\t%-5s", "---------------","--------------------", "----");
                rv.add(line);
            
            PreparedStatement s = mcn.prepareStatement("Select TID from TeamRoster where PID = ?");
            // Execute the query and store the results in a resultset object
            s.setString(1,pid);
            ResultSet r = s.executeQuery();
            r.next();
            String tid = r.getString(1);
            
            PreparedStatement s1 = mcn.prepareStatement("Select MID, MDate from MatchSchedule where Team1=? OR Team2=?") ;
            s1.setString(1, tid);
            s1.setString(2, tid);
            ResultSet r1 = s1.executeQuery();
            while(r1.next()){
                String mid = r1.getString(1);
                String mdate= r1.getString(2);
                
             PreparedStatement s2 = mcn.prepareStatement("Select Goals from MatchRoster where MID = ? AND PID =? ");
             s2.setString(1, mid);
             s2.setString(2, pid);
             ResultSet r2 = s2.executeQuery();
             
             int goals=0;
             while(r2.next()){
             goals = r2.getInt(1);
             }
             PreparedStatement s3 = mcn.prepareStatement("(Select Team1 as TeamID from MatchSchedule where Team2=? AND MID = ?)Union All (Select Team2 as TeamID from MatchSchedule where Team1=? AND MID = ?)");
             s3.setString(1, tid);
             s3.setString(2, mid);
             s3.setString(3, tid);
             s3.setString(4, mid);
             ResultSet r3 = s3.executeQuery();
             r3.next();
             String teamid = r3.getString(1);
             
             PreparedStatement s4 = mcn.prepareStatement("Select TName from Team where TID = ?");
             s4.setString(1, teamid);
             ResultSet r4 = s4.executeQuery();
             r4.next();
             String tname= r4.getString(1);
             
             line = String.format("%-15s\t%-10s\t%-5d", tname, mdate,goals);
                rv.add(line);
            }
            
         /*   PreparedStatement s1 = mcn.prepareStatement("Select Team1, Team2 From MatchSchedule where Team1 = ? OR Team2 = ?");
            s1.setString(1, tid);
            s1.setString(2, tid);
            ResultSet r1 = s1.executeQuery();
           while(r1.next()){
            String tid1= r1.getString(1);
            String tid2= r1.getString(2);
      //      System.out.println(tid1);
       //     System.out.println(tid2);
       //     System.out.println("tid -----" +tid);
            
            String t = null;
            if(tid.equals(tid1)){
                t= tid2;
            }
            else if (tid.equals(tid2)){
                t = tid1;
            }
       //     System.out.println("t ===" +t);
            PreparedStatement s2 = mcn.prepareStatement("Select TName,TID From Team where TID = ? ");
            s2.setString(1, t);
            ResultSet r2 = s2.executeQuery();
       //     System.out.println("before while r2");
            while(r2.next()){
       //         System.out.println("in while r2");
            String tname = r2.getString(1);
            String ti = r2.getString(2);
            
            PreparedStatement s3 = mcn.prepareStatement("Select MID, MDate from MatchSchedule where (Team1=? AND Team2=?) OR (Team1=? AND Team2 = ?)");
            s3.setString(1, tid);
            s3.setString(2, ti);
            s3.setString(3, ti);
            s3.setString(4, tid);
            ResultSet r3 = s3.executeQuery();
            while(r3.next()){
         //       System.out.println("before while r3");
            String mid  = r3.getString(1);
            String mdate = r3.getString(2);
            
            PreparedStatement s4 = mcn.prepareStatement("Select Goals from MatchRoster where MID=?");
            s4.setString(1, mid);
            //s4.setString(2, tid);
            //s4.setString(3, tid);
            ResultSet r4 = s4.executeQuery();
            while(r4.next()){
         //       System.out.println("before while r4");
                int goals = r4.getInt(1);
                line = String.format("%-15s\t%-10s\t%-5d", tname, mdate,goals);
                rv.add(line);
                //System.out.println(line);
           
            }
            }
            }
            }*/
           
            
            

            // now display the output
           /* ResultSetMetaData rsmd = rs.getMetaData();

            int ncols = rsmd.getColumnCount();
            String header = "";
            for (int i = 1; i <= ncols; i++) {
                header += rsmd.getColumnName(i) + "   ";
            }
            rv.add(header);*/
            

            

            // close everything
            st.close();

        } catch (SQLException ex) {
            ex.printStackTrace();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return rv;
        }
    
    public List<String> ListTeamPoints(String tid) {
        List<String> retval = new ArrayList<String>();
        try {
            
            PreparedStatement pst = mcn.prepareStatement("Select Tname from Team where TID = ?");
            pst.setString(1, tid);
            ResultSet res = pst.executeQuery();
            String line = String.format("%-12s\t%-3s\t%-3s\t%-3s\t%-3s\t%-3s\t%-3s\t","Team","W","T","L","G","A","P");
            retval.add(line);
            line = String.format("%-12s\t%-3s\t%-3s\t%-3s\t%-3s\t%-3s\t%-3s\t","------------","---","---","---","---","---","---");
            retval.add(line);
            
            while(res.next()){
                String name = res.getString(1);
                
                PreparedStatement pst1 =mcn.prepareStatement("Select Count(*) as Won from Match M1,Match M2 where M1.TID = ? AND M1.MID = M2.MID AND M1.Goals>M2.Goals ");
                pst1.setString(1, tid);
                ResultSet res1 = pst1.executeQuery();
                res1.next();
                    String won = res1.getString(1);
            //        System.out.print(won);
                
                PreparedStatement pst2 =mcn.prepareStatement("Select Count(*) as Ties from Match M1,Match M2 where M1.TID = ? AND M1.MID = M2.MID AND M1.Goals = M2.Goals AND M1.TID<>M2.TID ");
                pst2.setString(1, tid);
                ResultSet res2 = pst2.executeQuery();
                res2.next();
                    String ties = res2.getString(1);
              //      System.out.print(ties); 
                    
                PreparedStatement pst3 =mcn.prepareStatement("Select Count(*) as Lost from Match M1,Match M2 where M1.TID = ? AND M1.MID = M2.MID AND M1.Goals<M2.Goals ");
                pst3.setString(1, tid);
                ResultSet res3 = pst3.executeQuery();
                res3.next();
                    String lost = res3.getString(1);
                //    System.out.print(lost);
                
                PreparedStatement pst4 =mcn.prepareStatement("Select Sum(Goals) as Goals from TeamRoster where TID = ?; ");
                pst4.setString(1, tid);
                ResultSet res4 = pst4.executeQuery();
                res4.next();
                    String goals = res4.getString(1);
             //       System.out.print(goals); 
                    
                PreparedStatement pst5 =mcn.prepareStatement("Select sum(Goals) from MatchRoster where MID IN (Select MID from MatchSchedule where Team1 = ? OR Team2 = ?) AND\n" +
"PID IN (Select PID from TeamRoster where TID IN ((Select Team1 from MatchSchedule WHERE Team2 = ?)Union (Select Team2 from MatchSchedule where Team1 = ?)))");
                pst5.setString(1, tid);
                pst5.setString(2, tid);
                pst5.setString(3, tid);
                 pst5.setString(4, tid);
                
                ResultSet res5 = pst5.executeQuery();
                res5.next();
                    String goalsall = res5.getString(1);
            //        System.out.print(goalsall); 
                    
               
                    int points = (Integer.parseInt(won)*3) + (Integer.parseInt(ties)*1) ;
            //        System.out.print(points); 
                    
                    line = String.format("%-12s\t%-3s\t%-3s\t%-3s\t%-3s\t%-3s\t%-3s\t", name,won, ties,lost,goals,goalsall,points);
                    retval.add(line);
                    
            }
            
            
        } catch (SQLException ex) {
            ex.printStackTrace();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return retval;
        }
        
    
    
    
}